solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/HeuristicBoostedModel.java

```diff
@@ -0,0 +1,179 @@
+          type = BoostType.sum;
+        } else if ((!BoostType.sum.product.toString().equals(fetchedType))) {
+          throw new ModelException("Model " + HeuristicBoostedModel.this.getName() + " boost type : " + fetchedType + " is not supported");
+        }
+      }
+    }
+
+    private void checkParams(Map<String, Object> euristicBoostParams) {
+      checkParam(euristicBoostParams, FEATURE_NAME);
+      checkParam(euristicBoostParams, WEIGHT);
+    }
+
+    private void checkParam(Map<String, Object> featureBoostParams, String paramName) {
+      String modelName = HeuristicBoostedModel.this.getName();
+      if (!featureBoostParams.containsKey(paramName)) {
+        throw new ModelException("Model " + modelName + " required param " + paramName + " not defined");
+      }
+    }
+
+    private int getFeatureIndex( String featureName) {
+      String modelName = HeuristicBoostedModel.this.getName();
+      final List<Feature> features = getFeatures();
+      Feature featureToBoost = null;
+      for (Feature f : features) {
+        if (f.getName().equals(featureName)) {
+          featureToBoost = f;
+          break;
+        }
+      }
+
+      if (featureToBoost != null) {
+        return featureToBoost.getId();
+      } else {
+        throw new ModelException("Model " + modelName + " doesn't contain any feature with the name=[" + feature + "}");
+      }
+    }
+  }
+
+  public HeuristicBoostedModel(String name, List<Feature> features,
+                               String featureStoreName, List<Feature> allFeatures,
+                               NamedParams params) throws ModelException {
+    super(name, features, featureStoreName, allFeatures, params);
+    initHeuristicBoost();
+  }
+
+  protected void initHeuristicBoost() {
+    Map<String, Object> params = (Map<String, Object>) HeuristicBoostedModel.this.getParams().get(HEURISTIC_BOOST_PARAM);
+    if (params != null) {
+      this.heuristicFeatureBoost = new EuristicFeatureBoost(params);
+    }
+  }
+
+  @Override
+  public float score(float[] modelFeatureValuesNormalized) {
+    float internalModelScore = scoringAlgorithm.score(modelFeatureValuesNormalized);
+    float finalScore = internalModelScore;
+    if (heuristicFeatureBoost != null) {
+      float boostFeatureValue = modelFeatureValuesNormalized[heuristicFeatureBoost.featureIndex];
+      float weightedBoostFeatureValue = getWeightedBoostFeatureValue(internalModelScore, boostFeatureValue);
+
+      switch (heuristicFeatureBoost.type) {
+        case product:
+          finalScore *= weightedBoostFeatureValue;
+          break;
+
+        case sum:
+          finalScore += weightedBoostFeatureValue;
+          break;
+      }
+    }
+    return finalScore;
+  }
+
+  private float getWeightedBoostFeatureValue(float internalModelScore, float boostFeatureValue) {
+    float weightedBoostFeatureValue = heuristicFeatureBoost.weight * boostFeatureValue;
+    switch (heuristicFeatureBoost.type) {
+      case product:
```

Comment:
Should we assume the score is always positive ?
I got a lot of times negatives score with LambdaMART models.
That was a draft attempt to be compatible with boost score and lamdaMART score  in case the sign differs.

Though I didn't spend much time on that, indeed in case we have both negative boost and negative lambdaMART score it is not going to work.

Do you have any suggestion regarding that ?